packages/account/src/Components/account-limits/account-limits.tsx

```diff
@@ -12,366 +12,346 @@ import AccountLimitsFooter from './account-limits-footer';
-        payout: string | number;
-        lifetime_limit?: number;
-        market_specific: {
-            commodities: TAccountLimitsCollection[];
-            cryptocurrency: TAccountLimitsCollection[];
-            forex: TAccountLimitsCollection[];
-            indices: TAccountLimitsCollection[];
-            synthetic_index: TAccountLimitsCollection[];
-        };
-        num_of_days?: number;
-        num_of_days_limit: string | number;
-        remainder: string | number;
-        withdrawal_for_x_days_monetary?: number;
-        withdrawal_since_inception_monetary: string | number;
-    };
-    currency: string;
     footer_ref?: React.RefObject<HTMLElement>;
     is_app_settings?: boolean;
-    getLimits: () => Promise<{ data: object }>;
-    is_fully_authenticated: boolean;
-    is_from_derivgo?: boolean;
-    is_switching: boolean;
-    is_virtual: boolean;
     overlay_ref: HTMLDivElement;
     setIsOverlayShown?: (is_overlay_shown?: boolean) => void;
     setIsPopupOverlayShown?: (is_popup_overlay_shown: boolean) => void;
     should_bypass_scrollbars?: boolean;
     should_show_article?: boolean;
 };
 
-const AccountLimits = ({
-    account_limits,
-    currency,
-    footer_ref,
-    getLimits,
-    is_app_settings,
-    is_fully_authenticated,
-    is_switching,
-    is_virtual,
-    overlay_ref,
-    is_from_derivgo,
-    setIsOverlayShown: setIsPopupOverlayShown,
-    should_bypass_scrollbars,
-    should_show_article,
-}: TAccountLimits) => {
-    const isMounted = useIsMounted();
-    const [is_loading, setLoading] = React.useState(false);
-    const [is_overlay_shown, setIsOverlayShown] = React.useState(false);
-    const { is_appstore } = React.useContext(PlatformContext);
+const AccountLimits = observer(
+    ({
+        footer_ref,
+        is_app_settings,
+        overlay_ref,
+        setIsOverlayShown: setIsPopupOverlayShown,
+        should_bypass_scrollbars,
+        should_show_article = true,
+    }: TAccountLimits) => {
+        const { client, common } = useStore();
+        const { account_limits, currency, getLimits, is_fully_authenticated, is_virtual, is_switching } = client;
+        const { is_from_derivgo } = common;
+        const isMounted = useIsMounted();
+        const [is_loading, setLoading] = React.useState(false);
+        const [is_overlay_shown, setIsOverlayShown] = React.useState(false);
+        const { is_appstore } = React.useContext(PlatformContext);
 
-    React.useEffect(() => {
-        if (is_virtual) {
-            setLoading(false);
-        } else {
-            getLimits().then(() => {
-                if (isMounted()) setLoading(false);
-            });
-        }
-        // eslint-disable-next-line react-hooks/exhaustive-deps
-    }, []);
+        React.useEffect(() => {
+            if (is_virtual) {
+                setLoading(false);
+            } else {
+                getLimits().then(() => {
+                    if (isMounted()) setLoading(false);
```

Comment:
Do we need to call isMounted here as I've seen it's a ref with just a boolean set?

Reply:
I wouldn't make changes to existing logic because, I see the same behavior without it. But since this PR is related to replacing connect, I'll keep the change. Will investigate further as to why that hook is used in the APP